### DISCUSSION TITLE
Attempts to place text(s) using scripting doesn't work as intended

### DISCUSSION THREAD

**Discussion by UnforeseenOcean** [ImageMagick/ImageMagick#6257] [Help] [answered]
### ImageMagick version

7.1.0-51 Q16-HDRI

### Operating system, version and so on

Windows 10

### Description

Hello, I wanted to create images for my "Music to DVD" project, but for some reason, the text (caption) defies my command no matter what I try and I am at my wit's end. Please see the pictures for more information.

The script:
```
"bg-test.jpg" -size 1280x720 \( -page +300+300 ".\v2\512-The SIDH - Utopia.jpg" \) -background "rgba(255,255,255,0)" -fill "#fefefe"   -gravity Center -geometry +500-100 -size 720x290 -pointsize 60 -font "Carter-One" -interline-spacing -35 caption:"Questlords of Inverness\, Ride to the Galactic Fortress\!" -layers merge -write show:
```
Intended result (Please note that the "artist text" is not in the script):
![IMG1681620981](https://user-images.githubusercontent.com/11834016/232270505-28a3e990-c0d8-4796-96c1-c11243e8a12d.png)

Actual result: 
![magick-sBz4etgeIVrzqNukzECfQCWOmzgj5n9b](https://user-images.githubusercontent.com/11834016/232270680-a09e751e-7b19-4483-a548-d7f92a7b1546.jpg)

I must be doing something completely incorrect, but I don't know what is wrong with my script! Please help!

**Comment by snibgo** [comment, marked as answer]
"`-gravity centre`" makes your `caption:` text central within its image, ie "not justified". You want it left-justified, so the gravity should be off or NorthWest or West or SouthWest.

`-geometry` has no effect on "`-layers merge`". Instead, use `-page` as you did for the SIDH image, or `-repage`.

If that doesn't fix the problems, please post the inputs.

**Comment by UnforeseenOcean** [comment]
Using `-flatten` with a bunch of `-layer merge` did the trick:
```
#!/bin/env magick-script
"bg-test.jpg" -size 1280x720 \( -page +500+484 ".\v2\512-The SIDH - Utopia.jpg" -scale 50% \) \
-layers merge \
\( -page +540+230 -background "rgba(255,255,255,0)" -size 480x200 -fill "#fefefe" -gravity NorthWest -pointsize 40 -font "Carter-One" -interline-spacing -25 \
caption:"Questlords of Inverness\, Ride to the Galactic Fortress\!" \) \
-layers merge \
\( -page +540+415 -background "rgba(255,255,255,0)" -fill "#fefefe" -gravity NorthWest -pointsize 30 -font "Carter-One" -interline-spacing -35 \
caption:"Gloryhammer" \) -flatten \
-write show:
```
![magick-uTuGWLxDmkfDnBMkPsBRe4yIfxXDRGfD](https://user-images.githubusercontent.com/11834016/232319701-a73f7e6e-683d-41e7-b434-72c79ea3246e.jpg)
